core/src/main/scala/pureconfig/error/ConfigReaderFailure.scala

```diff
@@ -298,6 +299,16 @@ final case class NoValidCoproductChoiceFound(value: ConfigValue, location: Optio
 }
 
 /**
+ * A failure representing the inability to read a requested file.
+ */
+final case class CannotReadFile(path: Path) extends ConfigReaderFailure {
+  val location = None
+
+  def description = s"Unable to read file: ${path.toString}"
```

Comment:
I would prefer to say why the file is not readable. The message could be `"Unable to read file ${path.toString} because it doesn't exist"`